app/services/platforms/youtube.ts

```diff
@@ -229,39 +248,37 @@ export class YoutubeService extends StatefulService<IYoutubeServiceState>
     const request = new Request(url, { headers });
 
     return fetch(request)
-      .then(handlePlatformResponse)
+      .then(handleResponse)
       .then(response => this.userService.updatePlatformToken(response.access_token));
   }
 
-  putChannelInfo({ title, description }: IChannelInfo): Promise<boolean> {
-    this.SET_CHANNEL_INFO({ title, description });
-    return this.getLiveStreamId(false)
-      .then(() => this.fetchDescription())
-      .then(autopublishString => {
-        const headers = new Headers();
-        headers.append('Content-Type', 'application/json');
-
-        const fullDescription = new RegExp(autopublishString).test(description)
-          ? description
-          : autopublishString.concat(description);
-        const body = JSON.stringify({
-          snippet: {
-            title,
-            description: fullDescription,
-            scheduledStartTime: this.state.scheduledStartTime || new Date().toISOString(),
-          },
-          status: { privacyStatus: 'public' },
-          id: this.state.liveStreamId,
-        });
-        const endpoint = 'liveBroadcasts?part=snippet,status';
-        const method = this.state.liveStreamId ? 'PUT' : 'POST';
-
-        return platformRequest({
-          method,
-          body,
-          url: `${this.apiBase}/${endpoint}&access_token=${this.oauthToken}`,
-        }).then(() => true);
-      });
+  putChannelInfo(
+    { title, description }: IChannelInfo,
+    scheduledStartTime?: string,
+  ): Promise<boolean> {
+    return this.fetchDescription().then(autopublishString => {
+      const fullDescription = new RegExp(autopublishString).test(description)
+        ? description
+        : autopublishString.concat(description);
+      const endpoint = 'liveBroadcasts?part=snippet,status&default=true';
+      const data: Dictionary<any> = {
+        id: this.state.liveBroadcast.id,
+        snippet: {
+          title,
+          description: fullDescription,
+        },
+        status: { privacyStatus: 'public' },
+      };
+      if (scheduledStartTime) {
+        data.snippet.scheduledStartTime = scheduledStartTime;
```

Comment:
not sure why this is in a separate part of the code instead of assigning it in the object above

Reply:
`scheduledStartTime`  is an optional parameter. If you provide this param than Youtube considers this stream as scheduled. We don't need to provide `new Date().toISOString()` as we did before for the regular start streaming flow

Reply:
right but if the param is option when it's not passed in it'll be `undefined` and having a key be undefined in JS is the same as it not existing

Reply:
I see what you mean. That that will be omitted in JSON.stringify. That makes sense so